mitosheet/mitosheet/types.py

```diff
@@ -184,8 +195,22 @@ class FrontendFormulaReference(TypedDict):
     CodeSnippetEnvVars = Any # type:ignore
     FrontendFormulaString = Any # type:ignore
     FrontendFormulaReference = Any # type:ignore
+    FormulaLocationEntireColumn = Any # type:ignore
+    FormulaLocationToSpecificIndexLabels = Any # type:ignore
 
 
 FrontendFormulaPart = Union[FrontendFormulaString, FrontendFormulaReference]
 FrontendFormula = List[FrontendFormulaPart]
 
+FormulaAppliedToType = Union[FormulaLocationEntireColumn, FormulaLocationToSpecificIndexLabels]
+
+
+if sys.version_info[:3] > (3, 8, 0):
```

Comment:
TODO: cleanup and document types. Also, perhaps I can use literals here since it's 3.9?